libsolidity/formal/SMTEncoder.cpp

```diff
@@ -3216,3 +3251,30 @@ smt::SymbolicState& SMTEncoder::state()
 {
 	return m_context.state();
 }
+
+smtutil::Expression SMTEncoder::createSelectExpressionForFunction(
+	smtutil::Expression symbFunction,
+	std::vector<frontend::ASTPointer<frontend::Expression const>> const& args,
+	frontend::TypePointers const& inTypes,
+	unsigned long argsActualLength)
+{
+	std::vector<smtutil::Expression> symbArgs;
+	for (unsigned i = 0; i < argsActualLength; ++i)
```

Comment:
I think we can only assert the second part `inTypes.size() == argsActualLength`.
That's because `abi.decode`, from what I see in `visitABIFunction` we set the `argsActualLength` for it to 1, which may not be equal to the factual number of arguments.

Reply:
OK, so at lease `<=` for `args` then? So that we can be sure that we don't read past the array.